OsmAnd/src/net/osmand/plus/mapcontextmenu/other/TrackDetailsMenu.java

```diff
@@ -231,18 +237,24 @@ private QuadRect getRect(LineChart chart, float startPos, float endPos) {
 			} else {
 				float startDistance = startPos * dataSet.getDivX();
 				float endDistance = endPos * dataSet.getDivX();
-				for (WptPt p : segment.points) {
-					if (p.distance >= startDistance && p.distance <= endDistance) {
+				double previousSplitDistance = 0;
+				for (int i = 0; i < segment.points.size(); i++) {
+					if (i != 0) {
+						if (segment.points.get(i).distance < segment.points.get(i - 1).distance) {
+							previousSplitDistance += segment.points.get(i - 1).distance;
+						}
+					}
+					if (previousSplitDistance + segment.points.get(i).distance >= startDistance && previousSplitDistance + segment.points.get(i).distance <= endDistance) {
 						if (left == 0 && right == 0) {
-							left = p.getLongitude();
-							right = p.getLongitude();
-							top = p.getLatitude();
-							bottom = p.getLatitude();
+							left = segment.points.get(i).getLongitude();
```

Comment:
Why not p = segment.points.get(i); 
So many instractions every line (copy-paste)